### PR TITLE
Fork and fix GHA readme generator for local use [sc-73707]

### PR DESCRIPTION
**Pull request by lestroos**
I've forked the [github-action-readme-generator](https://github.com/bitflight-devops/github-action-readme-generator) and made a few changes to suit our own needs:

- Fixed the output component of the generator (update in `src/sections/update-outputs.ts`)
- Disabled commitlint -- the rules are pretty intrusive and was easier to disable rather than ignore everything
- Optionally use npm registry auth - we will use a locally generated package rather storing/distributing it via npm
- Add instructions for building and running the tool locally

~~I'll PR the outputs fix to the owner's source code and if/when that gets merged in, we can go back to using that version.~~ Update: there is an existing PR that covers the fix https://github.com/bitflight-devops/github-action-readme-generator/pull/179. It's been opened since Aug 2022 so we can probably consider the repo to be abandoned.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -57,6 +57,10 @@ inputs:
       Add a prefix to the README title
     required: false
     default: 'GitHub Action: '
+outputs:
+  test_test:
+    description: >
+      The test_test output is used to test the output component of the generator.
```
Comment (tay-bird) [inline action.yml]: Did you intend for this to be called test_test?

File: README.md [hunk 2]
```diff
@@ -29,6 +29,16 @@ This tool uses markdown comments as delimiting tokens within the `README.md` fil
...
 | --title_prefix, --prefix           | Add a prefix to the README title                                                            | [default: ""]    |
 
+### Build and Run Locally
+
+```sh
+yarn install
+npm install
+npm run build
+npm pack
+npx github-action-readme-generator-<version>.tgz
```
Comment (tay-bird) [inline README.md:39]: How can we integrate this into a pre-commit hook in other repositories? This PR is probably good to go with this content, but we should follow up with figuring that part out as well. I'll make a ticket for that.

Comment (lestroos) [reply README.md:39]: Sounds good, yes I figured CI integration would be a separate defined chunk of work.

File: action.yml (hunk 1 above)
Comment (lestroos) [reply action.yml]: Yep but I'll update this to just `test` for clarity.

**Review by tay-bird — APPROVED** [review]

**Pull request merged**